cmd/publishing-bot/main.go

```diff
@@ -183,6 +185,11 @@ func main() {
 					githubIssueErrorf("Failed to report logs on github issue: %v", err)
 					server.SetHealth(false, hash)
 				}
+				if strings.HasSuffix(err.Error(), memory.ErrRefHasChanged.Error()) {
+					// If the issue is just "reference has changed concurrently",
+					// then let us wait for 5 minutes and try again.
```

Comment:
Let's also add some logging here...also, can we add a comment here with a TODO to fix the actual flake?

Reply:
Can this end up in some weird recursion? :upside_down_face: 

We have seen this error consecutively several times...maybe we should let it retry only a certain number of times?

Reply:
done.

Reply:
i was thinking we should try this as-is and we can remove this if it does not work.